File: content/1-basics/5-while-loop/java/WhileLoop.java
import java.util.Scanner;

public class WhileLoop {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int number;

        System.out.println("Enter numbers (enter 0 to stop):");

        while (true) {
            System.out.print("Enter a number: ");
            number = scanner.nextInt();

            if (number == 0) {
                break;
            }

            System.out.println("You entered: " + number);
        }

        System.out.println("Loop terminated because you entered 0.");
        scanner.close();
    }
}

File: content/1-basics/5-while-loop/hs/while-loop.hs
main :: IO ()
main = do
    putStrLn "Enter numbers (enter 0 to stop):"
    loop

loop :: IO ()
loop = do
    putStr "Enter a number: "
    input <- getLine
    let number = read input :: Int
    if number == 0
        then putStrLn "Loop terminated because you entered 0."
        else do
            putStrLn $ "You entered: " ++ show number
            loop